### PR TITLE
Fix PorousFlowAddMaterialAction and PorousFlowAddMaterialJoiner

### LINKED ISSUE #13116: PorousFlow material actions not working with block restrictions
## Bug Description
The actions to automatically add PorousFlowJoiner materials as required don't work with block restricted materials that have a fluid phase dependency. Amazingly this hadn't been found until @WilkAndy hit it today.

## Steps to Reproduce
Simply add a block restriction to a material like PorousFlowRelativePermeability, sit back and watch the errors as they get printed to the screen.

## Impact
Stops users having block restricted materials, so should be fixed ASAP.

### PR DESCRIPTION
**Pull request by cpgr**
Enable these actions to work with block restricted materials.

Fixes #13116

### CODE REVIEW

**Comment by WilkAndy** [comment]
Thanks @cpgr !

File: modules/porous_flow/src/actions/PorousFlowAddMaterialJoiner.C [hunk 1]
```diff
@@ -65,6 +66,16 @@ PorousFlowAddMaterialJoiner::act()
         // Check if the material is evaluated at the nodes or qps
         const bool at_nodes = params.get<bool>("at_nodes");
 
+        // Get the vector of blocks that the material is restricted to
+        const std::vector<SubdomainName> block_ids =
+            params.get<std::vector<SubdomainName>>("block");
+
+        // String version of block id to use to ensure unique names
+        std::string block_str;
```
Comment (WilkAndy) [inline modules/porous_flow/src/actions/PorousFlowAddMaterialJoiner.C]: Did you want to have
```
std::string block_str = "";
```
or
```
const std::string block_str = (block_ids.empty() ? "" :  "_" + Moose::stringify(block_ids[0]));
```

**Review by WilkAndy — COMMENTED** [review]
Excellent, thanks so much @cpgr

File: modules/porous_flow/src/actions/PorousFlowAddMaterialJoiner.C (hunk 1 above)
Comment (cpgr) [reply modules/porous_flow/src/actions/PorousFlowAddMaterialJoiner.C]: I forgot to take these out after realising that I could add a single PorousFlowJoiner material that combined all block restricted materials.

File: modules/porous_flow/src/actions/PorousFlowAddMaterialJoiner.C [hunk 2]
```diff
@@ -28,7 +28,7 @@ validParams<PorousFlowAddMaterialJoiner>()
 }
 
 PorousFlowAddMaterialJoiner::PorousFlowAddMaterialJoiner(const InputParameters & params)
-  : Action(params)
+  : Action(params), _already_joined()
```
Comment (permcody) [inline modules/porous_flow/src/actions/PorousFlowAddMaterialJoiner.C]: You probably know this, but you don't need to explicitly initialize container types.

**Review by permcody — APPROVED** [review]

**Comment by WilkAndy** [comment]
Thanks for that review, @permcody .  Super surprising that we didn't pick up this bug earlier.

**Comment by cpgr** [comment]
Thanks Cody!

**Pull request merged**